lib/elixir/lib/function.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Function do
+  @moduledoc """
+  A set of funtions for working with functions.
+
+  We can have two types of captures functions - external and local -
+  external functions are our normal functions residing in modules while
+  local are the anonymous functions defined with `fn` or the capture
+  operator `&`.
+  """
+
+  @doc """
+  Captures a function by a given module, function name and arity.
+
+  Useful when you want to capture a function when having variables.
+
+  ## Examples
+
+    iex> Function.capture(String, :length, 1)
+    &String.length/1
+  """
+  @type information ::
+          :type
+          | :module
+          | :arity
+          | :name
+          | :env
+          | :pid
+          | :index
+          | :new_index
+          | :new_uniq
+          | :uniq
+  @type value :: any
+
+  @spec capture(atom, atom, integer) :: (... -> any)
+  def capture(mod, fun, arity) when is_atom(mod) and is_atom(fun) and is_integer(arity) do
+    :erlang.make_fun(mod, fun, arity)
+  end
+
+  @doc """
+  Returns a keyword list with information about a function.
+
+  The {key,value}s will include the following:
+    - :type - :local(for anonymous functions) or :external(for named functions)
+    - :module - an atom - the module where the function is defined when anonymous
+    or the module which the function refers to when it's a named one.
+    - :arity - the number of arguments the function is to be called with
+    - :name - the name of the functions
+    - :env - a list of the environment or free variables. For named functions, the returned list is always empty.
+
+  When it is an anonymous function it will also return info about:
+    - :pid - process identifier of the process that originally created the fun
```

Comment:
I wouldn't say fun but function. We need the trailing period in this sentence. I think sentence is a bit redundant. I would go with: PID of the process that created the function.